src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,435 @@
+            let executionResult = await util.executeAndThrowIfError(`az`, command.split(' '));
+            return executionResult.exitCode === 0;
+        } catch (err) {
+            githubActionsToolHelper.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided Container App Environment exists in the provided resource group.
+     * @param containerAppEnvironment - the name of the Container App Environment
+     * @param resourceGroup - the resource group that the Container App Environment is found in
+     * @returns true if the Container App Environment exists, false otherwise
+     */
+    public async doesContainerAppEnvironmentExist(containerAppEnvironment: string, resourceGroup: string): Promise<boolean> {
+        githubActionsToolHelper.debug(`Attempting to determine if Container App Environment with name "${containerAppEnvironment}" exists in resource group "${resourceGroup}"`);
+        try {
+            let command = `containerapp env show -n ${containerAppEnvironment} -g ${resourceGroup} -o none`;
+            let executionResult = await util.executeAndThrowIfError(`az`, command.split(' '));
+            return executionResult.exitCode === 0;
+        } catch (err) {
+            githubActionsToolHelper.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided resource group exists.
+     * @param resourceGroup - the name of the resource group
+     * @returns true if the resource group exists, false otherwise
+     */
+    public async doesResourceGroupExist(resourceGroup: string): Promise<boolean> {
+        githubActionsToolHelper.debug(`Attempting to determine if resource group "${resourceGroup}" exists`);
+        try {
+            let command = `group show -n ${resourceGroup} -o none`;
+            let executionResult = await util.executeAndThrowIfError(`az`, command.split(' '));
+            return executionResult.exitCode === 0;
+        } catch (err) {
+            githubActionsToolHelper.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Gets the default location for the Container App provider.
+     * @returns the default location if found, otherwise 'eastus2'
+     */
+    public async getDefaultContainerAppLocation(): Promise<string> {
+        githubActionsToolHelper.debug(`Attempting to get the default location for the Container App service for the subscription.`);
+        try {
+            let args = [`provider`, `show`, `-n`, `Microsoft.App`, `--query`, `resourceTypes[?resourceType=='containerApps'].locations[] | [0]`];
+            let executionResult = await util.executeAndThrowIfError(`az`, args);
+            // If successful, strip out double quotes, spaces and parentheses from the first location returned
+            return !executionResult.stderr ? executionResult.stdout.toLowerCase().replace(/["() ]/g, "").trim() : `eastus2`;
```

Comment:
Any reason why we check the error stream here rather than the exit code like we did for previous functions?

Reply:
No reason specifically, didn't think it would make a difference if we check the error stream or the exit code.
But looking at the Azure Devops task code, I think it makes sense to switch to exit code check just to be consistent.

Reply:
Updated.